docs/guide/component-testing.md

```diff
@@ -0,0 +1,402 @@
+})
+```
+
+#### Lit Components
+
+Install the community Lit testing package:
+
+```bash
+npm install --save-dev vitest-browser-lit
+```
+
+```ts
+// HelloWorld.ts
+import { html, LitElement } from 'lit'
+import { customElement, property } from 'lit/decorators.js'
+
+@customElement('hello-world')
+export class HelloWorld extends LitElement {
+  @property({ type: String })
+  name = 'World'
+
+  render() {
+    return html`<h1>Hello ${this.name}!</h1>`
+  }
+}
+
+declare global {
+  interface HTMLElementTagNameMap {
+    'hello-world': HelloWorld
+  }
+}
+```
+
+```ts
+// HelloWorld.test.ts
+import { expect, test } from 'vitest'
+import { render } from 'vitest-browser-lit'
+import { html } from 'lit'
+import './HelloWorld.js'
+
+test('renders name', async () => {
+  const screen = render(html`<hello-world name="Vitest"></hello-world>`)
+  const element = screen.getByText('Hello Vitest!')
+  await expect.element(element).toBeInTheDocument()
+})
+```
+
+## User Interactions
+
+Test user interactions using Vitest's locator methods:
+
+```js
+import { expect, test } from 'vitest'
+import { render } from 'vitest-browser-react'
+import { page } from '@vitest/browser/context'
+import LoginForm from './LoginForm'
+
+test('handles form submission', async () => {
+  render(<LoginForm />)
+
+  // Fill form fields
+  await page.getByLabelText(/username/i).fill('john_doe')
+  await page.getByLabelText(/password/i).fill('secret123')
+
+  // Submit form
+  await page.getByRole('button', { name: /submit/i }).click()
+
+  // Assert results
+  await expect.element(page.getByText('Login successful')).toBeInTheDocument()
+})
+```
+
+## Assertions
+
+Vitest provides DOM-specific assertions for component testing:
+
+```ts
+import { expect } from 'vitest'
+
+// Element presence
+await expect.element(getByText('Hello')).toBeInTheDocument()
+await expect.element(queryByText('Missing')).not.toBeInTheDocument()
+
+// Element properties
+await expect.element(getByRole('button')).toBeEnabled()
+await expect.element(getByRole('button')).toBeDisabled()
+
+// Text content
+await expect.element(getByRole('heading')).toHaveTextContent('Welcome')
+
+// Attributes
+await expect.element(getByTestId('link')).toHaveAttribute('href', '/home')
+
+// CSS classes and styles
+await expect.element(getByTestId('alert')).toHaveClass('error')
+await expect.element(getByTestId('box')).toHaveStyle({ color: 'red' })
+```
+
+## Testing Library Integration
+
+For frameworks not officially supported by Vitest, you can use Testing Library packages:
+
+```jsx
+// For Solid.js components
+import { render } from '@testing-library/solid'
+import { page } from '@vitest/browser/context'
+
+test('solid component test', async () => {
+  const { baseElement } = render(() => <MyComponent />)
+  const screen = page.elementLocator(baseElement)
+
+  await expect.element(screen.getByText('Hello')).toBeInTheDocument()
+})
+```
+
+## Node.js Component Testing
```

Comment:
This section feels out of place. While yes, you can write component tests in Node.js, nothing on this page can be used there (browser mode features are limited only to the browser mode, and we only describe them here)

The examples right after it only work in the browser mode too

Since most of this page only works in the browser mode, maybe it's better to move this section to the browser mode docs